app/models/case_log.rb

```diff
@@ -36,6 +36,9 @@ class CaseLog < ApplicationRecord
 
   validates_with CaseLogValidator
   before_save :update_status!
+  before_validation :process_postcode_changes!, if: :property_postcode_changed?
+  before_validation :reset_location_fields!, unless: :postcode_known?
+  before_validation :set_derived_fields!
```

Comment:
Now that these are set before validation we likely don't need to remove them from the fields checked by the status update (i.e. they don't need to be in the auto generated list now). Not important to do here though.